mission-control/docs/config-db/tutorials/file-scraper.md

```diff
@@ -0,0 +1,154 @@
+:::
+
+Now finally, we need to tell config-db how it can generate 3 different configs for each of those servers from one config file. We do that by using transformation scripts. In this example, we've used javascript but other scripting option are also available like - CEL Expression & Go Template.
+
+```yaml title='vms-scraper.yaml'
+apiVersion: configs.flanksource.com/v1
+kind: ScrapeConfig
+metadata:
+  name: vm-scraper
+  namespace: default
+spec:
+  schedule: '@every 30s'
+  file:
+    - type: $.role
+      id: $.ip
+      name: $.name
+      class: VirtualMachine
+      transform:
+        javascript: JSON.stringify(config.servers)
+      paths:
+        - /tmp/vms.json
+```
+
+To summarize, this scraper will scraper our file `/tmp/vms.json` every 30 seconds and generate 3 config items for each of the servers. If more servers are added/removed in future, they'll be reflected in config-db.
+
+## Run the scraper
+
+Save the scraper and wait for it to run. You should see the job status go green.
+
+![](../../images/tutorial-config-scrapers.png)
+
+You should now be able to see the 3 servers in Catalog.
+
+![](../../images/example-config-items-vms.png)
+
+The content of the config items should be exactly what we have in `/tmp/vms.json`.
+They are represented as YAML but internally they are still saved as JSON.
+
+![](../../images/example-vm-scraper-database.png)
+
+## Adding a new item
+
+In our `/tmp/vms.json`, let's add one more vm for a proxy server and see how config-db reacts to it.
+
+```json title=vms.json
+{
+  "servers": [
+    {
+      "name": "web-server",
+      "ip": "192.168.1.100",
+      "port": 80,
+      "role": "frontend"
+    },
+    {
+      "name": "proxy-server",
+      "ip": "192.168.1.103",
+      "port": 443,
+      "role": "proxy"
+    },
+    {
+      "name": "api-server",
+      "ip": "192.168.1.101",
+      "port": 8080,
+      "role": "backend"
+    },
+    {
+      "name": "database-server",
+      "ip": "192.168.1.102",
+      "port": 27017,
+      "role": "database"
+    }
+  ],
+  "logging": {
+    "level": "info",
+    "file_path": "/var/log/app.log"
+  }
+}
+```
+
+You should see a new config item for the proxy server.
+
+## Monitoring changes
+
+Let's modify the port of our api-server from `8080` to `3000`. Wait for a few seconds until the change is picked up.
+
+The catalog summary table now shows that there's 1 change in our api-server.
+![Catalog Overview Page](../../images/example-vm-scraper-changes-overview.png)
+
+To see the details of the change, click on the config item.
+![](../../images/example-vm-scraper-api-server-change.png)
+
+## Deleting items
+
+Remove the proxy-server that we added before. If you notice, it's not immediately removed by config-db. That is because config-db waits for a certain time before it deletes a config item that it doesn't see anymore. By default, the timeout is 30 minutes but it can be configured using the `--stale-timeout` flag.
```

Comment:
This is something that should go into installation instructions, with a link from here.  staleTimeout / retention.stale should be on the scrape config itself